core/meterer/meterer.go

```diff
@@ -41,13 +44,248 @@ func NewMeterer(
 	offchainStore OffchainStore,
 	logger logging.Logger,
 ) (*Meterer, error) {
-	// TODO: create a separate thread to pull from the chain and update chain state
-	return &Meterer{
+	ctx := context.Background()
+	m := Meterer{
 		Config: config,
 
 		ChainState:    paymentChainState,
 		OffchainStore: offchainStore,
 
 		logger: logger.With("component", "Meterer"),
-	}, nil
+	}
+	// goroutine to periodically refresh the on-chain state
+	go func() {
+		ticker := time.NewTicker(1 * time.Hour)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				if err := paymentChainState.RefreshOnchainPaymentState(ctx, nil); err != nil {
```

Comment:
is this thread safe?

Reply:
I think this is okay since when `RefreshOnChainPaymentState` swaps out the chain state's reservation and on-demand maps atomically, and individual queries (like this [one](https://github.com/Layr-Labs/eigenda/blob/67c0e1d0186b392087793f76a76a466cd4e7ad93/core/meterer/onchain_state.go#L17) takes the struct and not pointer. 
Behaviorally, if a refresh happened in the middle of a request in which the meterer is using an outdated on-chain state, we would need to explain it takes some time for the on-chain adjustment to be reflected. Please let me know if you have concerns about this, ideally I would want to avoid locks if they are not necessary

Reply:
I don't think `RefreshOnChainPaymentState` swaps out the maps atomically. It reads the onchain state and then updates its internal state, right?